File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/3082/BusinessApplication/BA.Web.10.0/Services/AuthenticationService.vb
Imports System.Security.Authentication
Imports System.ServiceModel.DomainServices.Hosting
Imports System.ServiceModel.DomainServices.Server
Imports System.ServiceModel.DomainServices.Server.ApplicationServices
Imports System.Threading
Namespace Web

    '' TODO: cambie a un extremo seguro al implementar la aplicación.
    ''       El nombre de usuario y la contraseña solo se deben pasar mediante https.
    ''       Para ello, establezca la propiedad RequiresSecureEndpoint de EnableClientAccessAttribute en true.
    ''   
    ''       <EnableClientAccess(RequiresSecureEndpoint:=True)>
    ''
    ''       En MSDN encontrará más información sobre el uso de https con un servicio de dominio.

    ''' <summary>
    ''' Servicio de dominio responsable de la autenticación de los usuarios cuando inician sesión en la aplicación.
    '''
    ''' La clase AuthenticationBase ya proporciona la mayor parte de la funcionalidad.
    ''' </summary>
    <EnableClientAccess()> _
    Public Class AuthenticationService
        Inherits AuthenticationBase(Of User)
    End Class
End Namespace

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1028/BusinessApplication/BA.Web/Services/AuthenticationService.cs
namespace $safeprojectname$
{
    using System.Security.Authentication;
    using System.ServiceModel.DomainServices.Hosting;
    using System.ServiceModel.DomainServices.Server;
    using System.ServiceModel.DomainServices.Server.ApplicationServices;
    using System.Threading;

    // TODO: 部署此應用程式時切換到安全端點。
    //       使用者的名稱和密碼只可以使用 https 傳遞。
    //       若要執行此動作，請將 EnableClientAccessAttribute 上的 RequiresSecureEndpoint 屬性設定為 true。
    //   
    //       [EnableClientAccess(RequiresSecureEndpoint = true)]
    //
    //       可以在 MSDN 上取得搭配  https 使用網域服務的詳細資訊。

    /// <summary>
    /// 網域服務負責在使用者登入應用程式時對使用者進行驗證。
    ///
    /// 大部分的功能已由 AuthenticationBase 類別提供。
    /// </summary>
    [EnableClientAccess]
    public class AuthenticationService : AuthenticationBase<User> { }
}
